tests/providers/google/cloud/hooks/test_dataprep.py

```diff
@@ -0,0 +1,95 @@
+from tenacity import RetryError
+
+from airflow.providers.google.cloud.hooks import dataprep
+
+JOB_ID = 1234567
+
+
+class TestGoogleDataprepHook(TestCase):
+    def setUp(self):
+        self.hook = dataprep.GoogleDataprepHook()
+
+    @patch("airflow.providers.google.cloud.hooks.dataprep.requests.get")
+    def test_mock_should_be_called_once(self, get_request_mock):
+        self.hook.get_jobs_for_job_group(JOB_ID)
+        get_request_mock.assert_called_once()
+
+    @patch(
+        "airflow.providers.google.cloud.hooks.dataprep.requests.get",
+        side_effect=[HTTPError(), mock.MagicMock()],
+    )
+    def test_raise_http_error_(self, get_request_mock):
+        self.hook.get_jobs_for_job_group(JOB_ID)
+        self.assertEqual(get_request_mock.call_count, 2)
+
+    @patch(
+        "airflow.providers.google.cloud.hooks.dataprep.requests.get",
+        side_effect=[
+            mock.MagicMock(),
+            HTTPError(),
+            HTTPError(),
+            HTTPError(),
+            HTTPError(),
+        ],
+    )
+    def test_should_not_retry_after_success(self, get_request_mock):
+        self.hook.get_jobs_for_job_group.retry.sleep = mock.Mock()  # pylint: disable=no-member
+        self.hook.get_jobs_for_job_group(JOB_ID)
+        self.assertEqual(get_request_mock.call_count, 1)
+
+    @patch(
+        "airflow.providers.google.cloud.hooks.dataprep.requests.get",
+        side_effect=[
+            HTTPError(),
+            HTTPError(),
+            HTTPError(),
+            HTTPError(),
+            mock.MagicMock(),
+        ],
+    )
+    def test_should_retry_after_error(self, get_request_mock):
+        self.hook.get_jobs_for_job_group.retry.sleep = mock.Mock()  # pylint: disable=no-member
+        self.hook.get_jobs_for_job_group(JOB_ID)
+        self.assertEqual(get_request_mock.call_count, 5)
+
+    @patch(
+        "airflow.providers.google.cloud.hooks.dataprep.requests.get",
+        side_effect=[HTTPError()],
+    )
+    def test_raise_stop_iteration_error_after_first_call(self, get_request_mock):  # pylint: disable=unused-argument
+        with self.assertRaises(RetryError) as cm:
+            self.hook.get_jobs_for_job_group.retry.sleep = mock.Mock()  # pylint: disable=no-member
+            self.hook.get_jobs_for_job_group(JOB_ID)
+        self.assertIsInstance(cm.exception.last_attempt.exception(), StopIteration)
+
+    @patch(
+        "airflow.providers.google.cloud.hooks.dataprep.requests.get",
+        side_effect=[HTTPError(), HTTPError(), HTTPError(), HTTPError(), HTTPError()],
+    )
+    def test_raise_http_error_after_five_calls(self, get_request_mock):
+        with self.assertRaises(RetryError) as cm:
+            self.hook.get_jobs_for_job_group.retry.sleep = mock.Mock()  # pylint: disable=no-member
+            self.hook.get_jobs_for_job_group(JOB_ID)
+        self.assertIsInstance(cm.exception.last_attempt.exception(), HTTPError)
+        self.assertEqual(get_request_mock.call_count, 5)
```

Comment:
Not sure if this is neccessary

Reply:
Why so? In this case I want to test if fifth call will be finished with success

Reply:
It was soooooooooooo bad day... So many mistakes. I am embarrassed. Do you think it is no necessary to leave this test case? I would leave it as it is. It shows that there are five retry calls and all of them are errors. But like as said, it was a really not a good day.

Reply:
No problem, you can leave it as there's no harm in having more assertions